### ISSUE TITLE
HTMLParser should ignore not native HTML attributes

### ISSUE THREAD

**Issue by alexanderniebuhr** [windicss/windicss#213]
Attributes not native to HTML (such in vue or react & attributes with non completly string values)
should be ignored by HTMLParser completly instead giving back as classes.

```html
<input :checked="isCheckbox ? value : undefined" />
<div class={isRead ? "red": "green"} />
```

running with following,
```
const parser = new HTMLParser(text);
const classes = parser.parseClasses();
```
 will give back the classes
```
isCheckbox ? value : undefined
isRead ? "red": "green"
```

**Comment by alexanderniebuhr** [comment]
another vue example
```html
:disabled="userSession == null"`
```

**Comment by antfu** [comment]
Should we ignore dynamic props in Vue and React like starting with `v-bind:`, `:` or `{...}` instead?

**Comment by alexanderniebuhr** [comment]
would not work for svelte props like.
```
<div class={isRead ? "red": "green"} />
```

so i suggest, ignore any prop starting with `v-bind:`, `:` and all normal props with either `={}` or `="{}"`
if i find more cases, i'll add here

**Comment by antfu** [comment]
Yes, that's exactly what I mean :P. We can have a block list and add more as we go.

**Comment by alexanderniebuhr** [comment]
actually all attributes with `w:...` e.g. `w:hover` should be included,
if it has only text `"..."` - if it has `={}` or `="{}"` also block list

**Comment by pikax** [comment]
Shouldn't it also sort the string in a `:class`? 

```html
<div :class="{ "bg-red-100 hover:cursor-pointer": isRed}">
<div :class="["bg-red-100 hover:cursor-pointer", isRed && "bg-red-100 hover:cursor-pointer"]"/>
```

**Comment by alexanderniebuhr** [comment]
@pikax yeah it should, but the issue here is, it is hard to find out what are class names and what are variable names..
it also would need to support al type of functions e.g. vue, svelte, react, template literals, spread operator.... 

if you have an idea how to handle, let us know!!

**Comment by pikax** [comment]
Not sure about with other frameworks but I would say it should be safe to get the `class` content and sort the strings inside, a simple regex can be done for that
```javascript
/'[^']*'/g
```

To prevent some unwanted string changes you can also check for the string content if it contains valid classnames
```ts
// input
<div :class="{ 'bg-red-100 hover:cursor-pointer' isRed}">
<div :class="['bg-red-100 hover:cursor-pointer', isRed && 'bg-red-100 hover:cursor-pointer']"/>
// regex matches
const classNames = 'bg-red-100 hover:cursor-pointer'

// it should only return if the classNames contains any valid tailwind-css class
function shouldOrder(classNames){
  return classNames.split(' ').some(isWindiCSS)
}

function isWindiCSS(className) {
 // check if is a valid one
}

```

I would say is safe to assume that all the strings inside of class attribute should be handled by windiCSS (if the user is using windiCSS).


On the template literals is a bit tricky because is a decision if you want to sort it or not, for example personally I would like windiCSS to be sorted inside of template literal, but probably not to mess around the `${}` content, eg

```ts
`bg-blue-100 hover:cursor-pointer ${isRed ? 'bg-red-100' : ''} text-green-500 hover:text-red-800`
// I would expect the string to be sorted in 3 parts
"bg-blue-100 hover:cursor-pointer" - left of `${}`
'bg-red-100' - content of string literal 
'text-green-500 hover:text-red-800' - right of `${}`
```

This would translate the same if you have many more.

**Comment by alexanderniebuhr** [comment]
might a good start. just some more examples of svelte stuff. do not know when we will attack this issue here, but when we do, we will try to allow sorting first .. if that breaks to much, we will go with the block list i suppose

```svelte
<!-- Empty template literal as expression -->
<div class={`red`}>Should be red!</div>
<!-- Passing expression prop in template -->
<div class="{myClass ? 'red' : ''}">Should be red!</div>
<!-- Template literal as expression -->
<div class={`${myClass}`}>Should be black!</div>
<!-- Yet another template literal as expression -->
<div class={`${myClass ? "red" : ""}`}>Should be red!</div>
<!-- Let's spice things up... -->
<div class={['px-1.5', myClass ? 'red' : ''].join(" ")}>Should be red!</div>
```

**Comment by antfu** [comment]
How is `check` get matched? I am seeing it only checks class attr

https://github.com/windicss/windicss/blob/d172c896201237bc89c6e6b067ba92b08ba652b9/src/utils/parser/html.ts#L9-L9

**Comment by alexanderniebuhr** [comment]
const parser = new HTMLParser(text);

**Comment by alexanderniebuhr** [comment]
@antfu you are totally right. naming confusing... we do have a "copy" version of HTMLParser hard coded in extension... the issue happens there ... not in the original HTMLParser.. i will fix this... should rename that
